src/components/BottomTabs/BottomTabs.js

```
@@ -0,0 +1,135 @@
+/* @flow */
+
+import React from 'react';
+import { StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import withTheme from '../../core/withTheme';
+import BottomNavigation, { Tab } from 'react-native-material-bottom-navigation';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import type { Theme } from '../../types/Theme';
+
+type TabType = {
+  /*
+   * Label for tab tab
+   */
+  label: string,
+
+  /*
+   * Name of the icon from Material Icons
+   * https://material.io/icons/
+   */
+  icon: string,
+};
+
+type Props = {
+  /*
+   * Wheter we color background of the navigation
+   * If true, primary color is used, while icons and ripple is white
+   * otherwise, color icons and ripple with primary color, keeping bg white
+   */
+  colorNavigationBackground: boolean,
+
+  /*
+   * Decides how tabs should look like
+   */
+  tabs: TabType[],
+
+  /*
+   * Invoked with `newTabIndex` and `oldTabIndex` when tab is switched
+   */
+  onTabChange: (newTabIndex: number, oldTabIndex: number) => mixed,
+  theme: Theme,
+};
+
+class BottomTabs extends React.Component<void, Props, void> {
+  static propTypes = {
+    tabs: PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.string,
+        icon: PropTypes.string,
+      })
+    ).isRequired,
+    colorNavigationBackground: PropTypes.bool,
+    onTabChange: PropTypes.func,
+    theme: PropTypes.object.isRequired,
+  };
+
+  componentDidMount() {
+    this._warnAboutTabNumber(this.props);
+  }
+  componentWillReceiveProps(nextProps) {
+    this._warnAboutTabNumber(nextProps);
+  }
+
+  /*
+   * We use Bottom Navigation only if tab number is between 3 and 5
+   */
+  _warnAboutTabNumber = ({ tabs }) => {
+    if (tabs.length < 3 || tabs.length > 5) {
+      throw new Error('Number of tabs in BottomTabs must be between 3 and 5.');
```

Why does it have to be from 3 to 5?

---

G designs says `Three to five top-level destinations`

---

I feel this should be `tabs.length > 5`. Because I think it is fine to use only 2. Moreover, not sure if we should throw an error or a warning say, hey! this might look bad if you use more than 5 options.